### ISSUE TITLE
Setting layer property inside OnCameraMoveListener causes infinite recursion

### ISSUE THREAD

**Issue by themics** [mapbox/mapbox-gl-native#10687]
**Platform: Android**
**Mapbox SDK version: 5.2.1**

### Steps to trigger behavior

 1. Change layer's fill color inside `OnCameraMoveListener`.
```java
map.addOnCameraMoveListener(() -> {
    FillLayer layer = mapboxMap.getLayerAs("water");
    int color = layer.getFillColor().getColorInt() == Color.RED ? Color.BLUE : Color.RED;
    layer.setProperties(fillColor(color));
});
````

### Expected behavior

- Layer's fill color is changing frame by frame.

### Actual behavior

- App crashed by infinite recursion.
- Even app does not fall to infinite recursion, `OnCameraMoveListener` called one more time.

Changing a layer's property triggers repaint, so `MapObserver::onCameraIsChanging()` / `onCameraDidChange()` is called on the same call stack.

**Comment by tobrun** [comment]
Thank you for reaching out and reporting this. Was able to reproduce. For now you can work around this by posting your callback code to the end of the message queue using `Handler#post(Runnable)`. I'm looking into fixing this up at the SDK side by introducing same setup there.

**Comment by tobrun** [comment]
Closed with #10690
